src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginUtil.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Agent.Plugins.PipelineArtifact
+{
+    public static class PipelineArtifactPathHelper
+    {
+        // This collection of invalid characters is based on the characters that are illegal in Windows/NTFS filenames.
+        // Also prevent files (pipeline artifact names) from containing "/" or "\" due to the added complexity this introduces for file pattern matching on download.
+        private static readonly char[] ForbiddenArtifactNameChars =
```

Comment:
I think Path.GetInvalidFileNameChars() should give the same result.

Reply:
In the content in the array changes or if we want to update this list, I think it's better for use to create our own ForbidenChars array